docs/proposals/helm-charts.md

```diff
@@ -0,0 +1,169 @@
+However, with [Helm](https://helm.sh) being the "package manager for Kubernetes",
+there is already a strong collection of applications defined in the form of
+Helm charts. Adding the ability to the broker to consume these Helm charts as
+Service Bundles allows work already done on Helm charts to be made available in
+a cluster.
+
+## Phase 1 - Adding a Basic Helm Chart Registry Adapter
+
+### Creating a Base Image for Executing Helm Charts
+
+Ansible Playbook Bundles (APBs) have already been described as an instance of
+the more generic Service Bundle. Here, we will be creating a new type of
+Service Bundle, Helm Bundles, that respect the Service Bundle contract but use
+Helm as the runtime. We already have a [Helm Bundle
+Base](https://github.com/ansibleplaybookbundle/helm-bundle-base) and the
+changes shown in [Helm Bundle Base PR 2](https://github.com/ansibleplaybookbundle/helm-bundle-base/pull/2)
+show the work required to implement the Service Bundle contract using Helm.
+
+### Broker Config Options
+
+It is expected that a broker admin who wants to expose Helm charts would update
+the broker's config to look something like:
+
+```
+registry:
+  - type: helm
+    name: stable
+    url: "https://kubernetes-charts.storage.googleapis.com"
+    base_image: "docker.io/djzager/helm-bundle-base:latest"
+    white_list:
+      - ".*"
+```
+
+1. Type: refers to the registry adapter to handle this registry
+1. Name: gives a name to this registry item
+1. URL: the URL for the Helm chart repository
+1. Base Image: the container image to use when provisioning/deprovisioning a
+   Helm chart
+1. White List/Black List: allows the Admin to filter out Helm charts
+
+#### Changes to the Registry Package
+
+- Add the ability to read the `base_image` from the broker config.
+- Pass the URL and BaseImage to the registry adapter.
+- Instantiate a Helm registry adapter to handle Helm Chart registries in the
+  config.
+
+#### Creating a Helm Adapter
+
+The `HelmAdapter` struct will look something like, the addition of `Charts`
+makes it possible to save our work and only ready the registries index file
+once:
+
+```
+// import "k8s.io/helm/pkg/repo"
+// HelmAdapter - Helm Registry Adapter
+type HelmAdapter struct {
+    Config Configuration
+    Log    *logging.Logger
+    Charts map[string]*repo.ChartVersion
+}
+```
+
+When the broker calls `GetImageNames()` the Helm adapter will read the
+`index.yaml` found at the Helm Chart Repository URL based on the [chart
+repository
+structure](https://github.com/kubernetes/helm/blob/master/docs/chart_repository.md#the-chart-repository-structure)
+and add the latest version of each chart to the `Charts` field as well as
+its name to the list of `imageNames` to be returned.
+
+After these image names are filtered, the subsequent call to `FetchSpecs()`
+will find the chart by name in the `Charts` field and create a `Spec` object
+for it. For example:
```

Comment:
So will a call to `bootstrap` ever regenerate this `Charts` cache? How will we pick up changes?

Reply:
To my understanding, yes. Every call to `bootstrap` should create a new instance of a `HelmAdapter` for each of the `helm` type config entries, calling `GetImageNames()` and `FetchSpecs()` again.

Reply:
I don't think that is correct. Registries are made with an adapter as an unexposed field and then is used for subsequent calls. I do not think that bootstrap by itself is creating new adapters. Are you planning on adding some other behavior?

Reply:
Not intending to change the behavior. This concern should be overcome by rebuilding the list of image names and charts on each call to `GetImageNames()`, for example:

```
// GetImageNames - retrieve the images
func (r *HelmAdapter) GetImageNames() ([]string, error) {
        var imageNames []string

        r.Charts = map[string][]*repo.ChartVersion{}
```

Then, the subsequent call to  `FetchSpecs(image_names)` will be operating on the newly built list of image names and charts.